pkg/xmtp/listener.go

```diff
@@ -175,6 +175,7 @@ func (l *Listener) processEnvelope(env *v1.Envelope) error {
 				zap.Any("message_context", request.MessageContext),
 				zap.Bool("subscription_has_hmac_key", request.Subscription.HmacKey != nil),
 			)
+			return nil
```

Comment:
Need to return nil here otherwise it falls down into line 180 and still delivers the message even with shouldDeliver false.

Reply:
Oof. Thanks for fixing